backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardcarousel/BpkCardCarousel.kt

```diff
@@ -0,0 +1,161 @@
+import net.skyscanner.backpack.compose.card.BpkCard
+import net.skyscanner.backpack.compose.card.BpkCardCorner
+import net.skyscanner.backpack.compose.card.BpkCardPadding
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicator
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicatorStyle
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import kotlin.math.absoluteValue
+import kotlin.math.max
+
+@SuppressLint("RestrictedApi")
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun BpkCardCarousel(
+    cards: List<@Composable BoxScope.() -> Unit>,
+    modifier: Modifier = Modifier,
+    currentCard: Int = 0,
+    onCardChange: ((Int) -> Unit)? = null,
+) {
+    val totalCard by remember { mutableIntStateOf(cards.size) }
+    val pagerState = rememberPagerState(pageCount = { getModdedPageCount(totalCard) })
+
+    LaunchedEffect(pagerState) {
+        pagerState.scrollToPage(getModdedCurrentPageNumber(currentCard, totalCard))
+        snapshotFlow { pagerState.currentPage }.collect { page ->
+            onCardChange?.invoke(page % totalCard)
+        }
+    }
+
+    BoxWithConstraints(modifier = modifier) {
+        val scope = this
+        val itemWidth = scope.maxWidth.value * CARD_WIDTH_PERCENTAGE
+        val contentPadding = max(0f, (scope.maxWidth.value - itemWidth) / 2)
+
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            HorizontalPager(
+                modifier = Modifier.testTag("pager").weight(1f),
+                state = pagerState,
+                contentPadding = PaddingValues(horizontal = contentPadding.dp),
+            ) { page ->
+                Box(modifier = Modifier
+                    .width(itemWidth.dp)
+                    .graphicsLayer {
+                        val pageOffset =
+                            ((pagerState.currentPage - page) + pagerState.currentPageOffsetFraction).absoluteValue
+                        with(lerp(UNFOCUSED_CARD_SCALE, 1f, 1f - pageOffset.coerceIn(0f, 1f))) {
+                            scaleY = this
+                            scaleX = this
+                        }
+                    }) {
+                    cards[page % totalCard]()
+                }
+            }
+            if (totalCard > 1) {
+                BpkPageIndicator(
+                    modifier = Modifier.testTag("pageIndicator"),
+                    totalIndicators = totalCard,
+                    currentIndex = pagerState.currentPage % totalCard,
+                    style = BpkPageIndicatorStyle.Default,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun BpkCarouselCard(
+    imageAccessibilityLabel: String,
```

Comment:
we generally use `contentDescription` on android to match the platform

Reply:
Updated